src/webnn/native/dml/UtilsDML.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 The WebNN-native Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef WEBNN_NATIVE_UTILSDML_H_
+#define WEBNN_NATIVE_UTILSDML_H_
+
+#include <deque>
+#include <map>
+#include <vector>
+
+#include "common/Assert.h"
+#include "common/Log.h"
+
+namespace webnn::native::dml {
+
+#define RETURN_IF_FAILED(EXPR)                                                                    \
```

Comment:
nit: macros do NOT scope to namespace. It's not a bad idea to prefix them (eg. `WEBNN_*`) to avoid future compilation-time conflicts.

Reply:
OK， and do you mean rename as `WEBNN_RETURN_IF_FAILED` which seems a little long?

Reply:
Alternatively, we could use a template (this keeps it under the namespace). Otherwise, `WEBNN_RETURN_IF_FAILED` seems fine as well.

Reply:
Use `WEBNN_RETURN_IF_FAILED `currently.